jobmon_core/src/jobmon/distributor/task_instance_batch.py

```diff
@@ -109,6 +109,19 @@ def log_distributor_ids(self) -> None:
             app_route=app_route, message=data, request_type="post"
         )
 
+    def transition_to_killed(self) -> None:
+        """Mark all TIs in this batch as killed in the DB (ERROR_FATAL)."""
+        # 1) Make an HTTP call to the server if you have a relevant endpoint
+        #    that transitions them to a 'killed' or ERROR_FATAL state in bulk.
+        app_route = f"/array/{self.array_id}/transition_to_killed"
+        data = {
+            "batch_number": self.batch_number,
+            "reason": "Self-killed",
```

Comment:
Is the "reason" parameter actually being used in the transition_to_killed route?